tablo/models.py

```diff
@@ -156,26 +182,25 @@ def _get_time_extent(self):
 
     @property
     def fields(self):
-        fields = []
-        with connection.cursor() as c:
-            c.execute('select column_name, is_nullable, data_type from information_schema.columns where table_name = %s;', [self.table])
-            # c.description won't be populated without first running the query above
-            for field_info in c.fetchall():
-                field_type = field_info[2]
-                if field_info[0] == 'db_id':
-                    field_type = 'OID'
-                elif field_info[0] == POINT_FIELD_NAME:
-                    field_type = 'Geometry'
-
-                fields.append({
-                    'name': field_info[0],
-                    'alias': field_info[0],
-                    'type': POSTGIS_ESRI_FIELD_MAPPING[field_type],
-                    'nullable': True if field_info[1] == 'YES' else False,
-                    'editable': True
-                })
-
-        return fields
+        if self._fields is None:
+            fields = get_fields(self.table)
+            for field in fields:
+                if field['name'] == 'db_id':
+                    field['type'] = 'esriFieldTypeOID'
+                elif field['name'] == POINT_FIELD_NAME:
+                    field['type'] = 'esriFieldTypeGeometry'
+            self._fields = fields
+
+        return self._fields
+
+    @property
+    def related_fields(self):
+        if self._related_fields is None:
+            self._related_fields = {
```

Comment:
After thinking about this a while, I came up with the following:
```python
fields_by_title = ((r.related_title, f) for r in self.relations for f in r.fields)

self._related_fields = {}
    for table, field in fields_by_title:
        field_key = '{0}.{1}'.format(table, field['name'])
        self.related_fields[field_key] = field
```
I hope this suffices: it preserves most of the performance and makes it more readable.